src/django_components/perfutil/component.py

```diff
@@ -0,0 +1,169 @@
+import re
```

Comment:
In this file I implemented the top-down approach to rendering nested components, as desribed in[ Idea 2 - Render components top-down instead of bottom-up](https://github.com/django-components/django-components/issues/14#issuecomment-2596096632).

I've put this into a separate file, so I could proprly document and explain what's going on, and to keep the `component.py` file relatively navigatable, since it's already quite big.

Please see the comments throughout this file for details.